Create extract_overlapped_patches

EzicStar opened the pull request.

EzicStar committed:
Create extract_overlapped_patches

martingra commented on extract_overlapped_patches.py:
```diff
@@ -0,0 +1,87 @@
+    elif img_width < patch_width:
+        raise ValueError("Patch width can't be bigger than image width")
+
+    cy = [] # List of the y-axis coordinates of each patch centre
```
Se podrá extraer esto en una función que sea usable tanto por x como por y? pareciera que se hace exactamente lo mismo en los dos pedazos de código, por lo que quizas recibiendo el img_height como parametro (size) y el h_overlap (overlap) se pueda devolver el cy (patch_centers)

martingra replied:
Digo, de línea 25 a 46

EzicStar replied:
Quedo mas fiero jajaja

EzicStar committed:
Update extract_overlapped_patches.py

- Made a generic function for calculating patch center coordinates
- Function description updated
- Removed unused library

martingra commented on extract_overlapped_patches.py:
```diff
@@ -0,0 +1,74 @@
+                cv2.imwrite(to_path + '/' + f'patch_{j}_{k}.png', patch_img)
+
+    # The used overlap proportion used in X and Y axis
+    return true_h_overlap/patch_height, true_w_overlap/patch_width
```
Se podrán agregar las imágenes como salida en vez de guardarlas en el disco? así las podemos usar para llamar a yolo directamente de lo que tenemos en memoria.

Quizás un diccionario donde la key sea la coordenada x_y y el value la imagen de opencv

EzicStar committed:
Changed return type

- Function now returns a dictionary with a tuple of XY coordinates of the patch center as a key, and the patch image as a value.
- Save boolean parameter added to choose if you want to save on disk the patches
- Description updated
